apps/server/src/entities/learnroom/course.entity.ts

```diff
@@ -3,7 +3,7 @@ import { ObjectId } from '@mikro-orm/mongodb';
 import { BaseEntityWithTimestamps, EntityId } from '@shared/domain';
 
 interface ICourseProperties {
-	name: string;
+	name?: string;
```

Comment:
why is the course repository not registered in the entity manager (decorator at repository or course entity)? why is an own baserepository used instead of the one of mikroorm?

Reply:
sorry i don't understand exacly what you mean...

Reply:
i mean the customRepository in entitydecorator or the entity relation on the repository introduced in https://mikro-orm.io/docs/repositories/#custom-repository

Reply:
that is mostly because a custom repository (as far as i understand) should always be derived from `EntityRepository<T>`:
https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/entity/EntityRepository.ts
that would provide quite some basic repository methods which is basically a good thing. but i thought we decided to build our very own repo services which basically wrap the entity manager and provide only the methods for the use cases.
otherwise i would really love to discuss the alternative of using `EntityRepository` as base class.

Reply:
discuss tomorrow